fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java

```diff
@@ -44,15 +42,31 @@ public RequestInterceptor requestInterceptor() {
 
     private void fillSecurityHeaders(RequestTemplate requestTemplate) {
         Instant instant = Instant.now();
-        requestTemplate.header(X_REQUEST_SIGNATURE, calculateSignature(requestTemplate.request(), instant));
+        requestTemplate.header(X_REQUEST_SIGNATURE, calculateSignature(requestTemplate, instant));
         requestTemplate.header(FINTECH_ID, tppProperties.getFintechID());
         requestTemplate.header(X_TIMESTAMP_UTC, instant.toString());
     }
 
-    private String calculateSignature(Request request, Instant instant) {
-        String xRequestId = request.headers().get(X_REQUEST_ID).stream().findFirst().orElse(null);
-        String operationType = request.headers().get(X_OPERATION_TYPE).stream().findFirst().orElse(null);
-        DataToSign dataToSign = new DataToSign(UUID.fromString(xRequestId), instant, OperationType.valueOf(operationType));
-        return requestSigningService.signature(dataToSign);
+    private String calculateSignature(RequestTemplate requestTemplate, Instant instant) {
+        String operationType = requestTemplate.headers().get(HttpHeaders.X_OPERATION_TYPE).stream().findFirst()
+                                       .orElseThrow(() -> new IllegalStateException(HttpHeaders.X_OPERATION_TYPE + MISSING_HEADER_ERROR_MESSAGE));
+        FeignTemplateToDataToSignMapper mapper = new FeignTemplateToDataToSignMapper();
+
+        switch (operationType) {
+            case "AIS":
+                if (requestTemplate.path().contains("/transactions")) {
+                    return requestSigningService.signature(mapper.mapToListTransactions(requestTemplate, instant));
+                }
+                return requestSigningService.signature(mapper.mapToListAccounts(requestTemplate, instant));
+            case "BANK_SEARCH":
```

Comment:
It makes sense to have class
```
enum RequestOperationType {
     BANK_SEARCH, AIS;

     public static RequestOperationType fromRequestPath(String path) {
                if (requestTemplate.path().contains("/transactions")) {
                    return AIS;
                }
     }

}
}
```

Reply:
OperationType enum was extended, isTransactionsPath and isBankSearchPath were added